### PR TITLE
Add default signing key as a generic verification method

### LINKED ISSUE #195: [Task] Rename Authentication Key and Chain
## Description
Describe the task, this may be scoped beyond a single PR.

Instead of using the authentication field and key for DID messages, utilize a general verification method. As this removes the logic behind the authentication chain name, I request a rename to the Integration chain (Also in the code). The name of the key we use for signing DID messages may be anything, but we can default to #_sign-x, where x is the number of the key, incremented every time it is updated. This should not be enforced though. 

## Motivation
Describe the motivation for the feature to be developed.

We currently always use an authentication key for signing and updating DID messages. This doesn't fall in line with the standardized way of using the authentication verifiable relationship according to the DID standard. Instead, we should utilize a general verification method.

## Resources
Link to any resources relevant for the task such as Issues, PRs, reference implementations, or specifications.

## To-do list
Create a task-specific to-do list . Please link PRs that match the To-do list item behind the item after it has been submitted.

- [x] Rename AuthChain and other references to IntChain (Integration Chain) in the code
- [ ] Remove enforcement of signing with "auth key"
- [ ] Edit examples to add verification method with a #_sign-1 key.
- [ ] Edit account to standardize the #_sign-x naming instead of auth keys.
- [x] Edit documentation to rename all references to AuthChain

## Change checklist
Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [ ] The feature or fix is implemented in Rust and across all bindings whereas possible.
- [ ] The feature or fix has sufficient testing coverage
- [ ] All tests and examples build and run locally as expected
- [ ] Every piece of code has been document according to the documentation guidelines.
- [ ] If conceptual documentation (mdbook) and examples highlighting the feature exist, they are properly updated.
- [ ] If the feature is not currently documented, a documentation task Issue has been opened to address this.

### PR DESCRIPTION
**Pull request by PhilippGackstatter**
# Description of change

- Rename the last remaining occurrences of auth chain to int chain. This is a breaking change because multiple method names have changed.
- Add the default verification method we use to sign document update messages as a generic verification method, instead of only as an [authentication](https://www.w3.org/TR/did-core/#authentication) method. However, the `authentication` verification relationship is attached to the method. An example of a newly generated did document thus looks like:

```
{
    "id": "did:iota:dev:7uUb2ES6qnuzsKjWG7Ft8iVz5Q7S3ZgXrcJGTiKskBi8",
    "verificationMethod": [
        {
            "id": "did:iota:dev:7uUb2ES6qnuzsKjWG7Ft8iVz5Q7S3ZgXrcJGTiKskBi8#_sign-0",
            "controller": "did:iota:dev:7uUb2ES6qnuzsKjWG7Ft8iVz5Q7S3ZgXrcJGTiKskBi8",
            "type": "Ed25519VerificationKey2018",
            "publicKeyBase58": "4GeSr3ZmRGDQiSmDni2Su5MZXm4gzgTmhCrxNybWAquy"
        }
    ],
    "authentication": [
        "did:iota:dev:7uUb2ES6qnuzsKjWG7Ft8iVz5Q7S3ZgXrcJGTiKskBi8#_sign-0"
    ],
    "created": "2021-09-28T11:53:47Z",
    "updated": "2021-09-28T11:53:47Z"
}
```

## Links to any relevant issues

fixes #195

## Type of change
Add an `x` to the boxes that are relevant to your changes.

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] Enhancement (a non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

## How the change has been tested

Change existing tests appropriately to test the changed functionality.

## Change checklist
Add an `x` to the boxes that are relevant to your changes.

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: identity-iota/src/did/doc/iota_document.rs [hunk 1]
```diff
@@ -113,9 +114,12 @@ impl IotaDocument {
   ///
   /// This must be guaranteed safe by the caller.
   pub unsafe fn from_authentication_unchecked(method: IotaVerificationMethod) -> Self {
+    let verification_method_did: DID = method.id().as_ref().clone();
+
     CoreDocument::builder(Default::default())
       .id(method.controller().clone().into())
-      .authentication(method)
+      .verification_method(method.into())
+      .authentication(MethodRef::Refer(verification_method_did))
```
Comment (cycraig) [inline identity-iota/src/did/doc/iota_document.rs:122]: Should we add a unit test to check the structure of the generated document, i.e. that it now has a single verification method which the authentication section references? The current tests use `authentication()` which resolves the reference and doesn't check for a verification method.

**Review by cycraig — APPROVED** [review]
I'm not a big fan of using `int` instead of `integration` since the latter is used throughout the rest of the projects, but I'm also not strongly opposed.

Adding one or two unit tests for the new default document structure may be appropriate. 
Otherwise looks good to me!

File: identity-iota/src/did/doc/iota_document.rs (hunk 1 above)
Comment (PhilippGackstatter) [reply identity-iota/src/did/doc/iota_document.rs:122]: I've added a test that checks if the verification method with the default fragment `#authentication` is the same as what `document.authentication()` resolves to. `IotaDocument` doesn't expose access to the set of `authentication` methods, and I didn't find a way to get a reference to the `CoreDocument` inside. But I think that test should be good enough.

Comment (cycraig) [reply identity-iota/src/did/doc/iota_document.rs:122]: I think one can check that the authentication is a reference as follows:

```rust
    match document.document.authentication().first().unwrap().clone().into_inner() {
      MethodRef::Refer(did) => assert_eq!(did.fragment().unwrap_or_default(), "#authentication"),
      MethodRef::Embed(_) => panic!("authentication method should be a reference"),
    }
```

**Pull request merged**